wp-content/plugins/core/src/Blocks/Types/Hero/Model.php

```diff
@@ -0,0 +1,134 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Blocks\Types\Hero;
+
+use Tribe\Project\Blocks\Types\Base_Model;
+use Tribe\Project\Templates\Components\Content_Block;
+use Tribe\Project\Templates\Models\Image;
+use Tribe\Project\Theme\Config\Image_Sizes;
+
+class Model extends Base_Model {
+
+	public function get_data() {
+		return [
+			'layout'            => $this->get_layout(),
+			'media'             => $this->get_media(),
+			'content'           => $this->get_content(),
+			'container_classes' => [ 'b-hero__container', 'l-container' ],
+			'media_classes'     => [ 'b-hero__media' ],
+			'content_classes'   => [ 'b-hero__content' ],
+			'classes'           => [ 'c-block', 'c-block--full-bleed', 'b-hero' ],
+			'attr'              => [],
+		];
+	}
+
+	/**
+	 * @return bool|mixed
+	 */
+	private function get_layout() {
+		return $this->get( Hero::LAYOUT, Hero::LAYOUT_LEFT );
+	}
+
+	/**
+	 * @return array
+	 */
+	private function get_media(): array {
+		if ( ! $this->get( Hero::IMAGE ) ) {
+			return [];
+		}
+
+		return $this->get_image( $this->get( Hero::IMAGE ) );
+	}
+
+	/**
+	 * @param $attachment_id
+	 *
+	 * @return array
+	 */
+	private function get_image( $attachment_id ): array {
+		return [
+			'attachment'    => Image::factory( (int) $attachment_id ),
+			'as_bg'         => true,
+			'use_lazyload'  => true,
+			'wrapper_tag'   => 'div',
+			'wrapper_class' => [ 'b-hero__figure' ],
+			'image_classes' => [ 'b-hero__img', 'c-image__bg' ],
+			'src_size'      => Image_Sizes::CORE_FULL,
+			'srcset_size'   => [
+				Image_Sizes::CORE_FULL,
+				Image_Sizes::CORE_MOBILE,
+			],
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	private function get_content(): array {
```

Comment:
I'm a little uncomfortable with all of the `get_*` method names, because they are not getting the thing, _per se_, but the arguments that will be used to build the thing. E.g., `get_content_args()` seems more accurate than `get_content()`, and properly distinguishes its behavior from `get_layout()`, which is returning the actual layout value, not args to build a layout.